rust/scx_stat/src/stat.rs

```diff
@@ -0,0 +1,262 @@
+                    && path.segments[1].ident == "vec"
+                    && path.segments[2].ident == "Vec"
+            }
+            _ => false,
+        };
+
+        if !is_vec {
+            return Ok(None);
+        }
+
+        if let PathArguments::AngleBracketed(ab) = &path.segments.last().unwrap().arguments {
+            let args = &ab.args;
+            if args.len() < 1 {
+                return Err(Error::new(args.span(), "T generic argument missing"));
+            }
+
+            match &args[0] {
+                GenericArgument::Type(ty) => Ok(Some(Self::Array(ScxStatKind::new(&ty, paths)?))),
+                _ => Ok(None),
+            }
+        } else {
+            Ok(None)
+        }
+    }
+
+    fn new_dict(path: &Path, paths: &mut BTreeMap<String, Path>) -> syn::Result<Option<Self>> {
+        if path.leading_colon.is_some() {
+            return Ok(None);
+        }
+
+        let is_btree_map = match path.segments.len() {
+            1 => path.segments[0].ident == "BTreeMap",
+            3 => {
+                path.segments[0].ident == "std"
+                    && path.segments[1].ident == "collections"
+                    && path.segments[2].ident == "BTreeMap"
+            }
+            _ => false,
+        };
+
+        if !is_btree_map {
+            return Ok(None);
+        }
+
+        if let PathArguments::AngleBracketed(ab) = &path.segments.last().unwrap().arguments {
+            let args = &ab.args;
+            if args.len() < 2 {
+                return Err(Error::new(args.span(), "K, V generic arguments missing"));
+            }
+
+            match (&args[0], &args[1]) {
+                (GenericArgument::Type(ty0), GenericArgument::Type(ty1)) => {
+                    let kind0 = ScxStatKind::new(&ty0, paths)?;
+                    let kind1 = ScxStatKind::new(&ty1, paths)?;
+
+                    if kind0.can_be_dict_key() {
+                        Ok(Some(Self::Dict {
+                            key: kind0,
+                            datum: kind1,
+                        }))
+                    } else {
+                        Err(Error::new(ty0.span(), "K must be an integer or String"))
+                    }
+                }
+                _ => Ok(None),
+            }
+        } else {
+            Ok(None)
+        }
+    }
+
+    pub fn new(ty: &Type, paths: &mut BTreeMap<String, Path>) -> syn::Result<Self> {
+        let kind = ScxStatKind::new(ty, paths)?;
+        if let ScxStatKind::Struct(_) = &kind {
+            if let Type::Path(path) = ty {
+                if let Some(ar) = Self::new_array(&path.path, paths)? {
+                    return Ok(ar);
+                }
+                if let Some(dict) = Self::new_dict(&path.path, paths)? {
+                    return Ok(dict);
+                }
+            }
+        }
+        Ok(Self::Datum(kind))
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub enum ScxStatAttr {
```

Comment:
what are your thoughts on having a set of additional tags/labels (ex [prometheus](https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels)). I guess the server/client implementations could add any additional tags downstream.